chia/simulator/block_tools.py

```diff
@@ -944,11 +943,12 @@ def get_consecutive_blocks(
             self.log.info(
                 f"Sub slot finished. blocks included: {blocks_added_this_sub_slot} blocks_per_slot: "
                 f"{(len(block_list) - initial_block_list_len)/sub_slots_finished}"
+                f"Sub Epoch Summary Included: {sub_epoch_summary is not None} "
             )
             blocks_added_this_sub_slot = 0  # Sub slot ended, overflows are in next sub slot
 
-            # Handle overflows: No overflows on new epoch
-            if new_sub_slot_iters is None and num_empty_slots_added >= skip_slots and new_difficulty is None:
```

Comment:
this is the actual bugfix, as there should really not be any overflows on a sub-epoch either. and new difficulty is only set once per epoch.